### ISSUE TITLE
[FR] Build options shall be available to application writers

### ISSUE THREAD

**Issue by jeandube** [Haivision/srt#2610]
build options affecting the public headers shall be set in a headers for apps.
Options such as ENABLE_AEAD_API_PREVIEW requires the app to define it whenever srt.h is included and this define is not automatically update whenever libsrt is updated/reconfigured/rebuilt.
```
<snip>
   SRTO_RETRANSMITALGO = 61, // An option to select packet retransmission algorithm
#ifdef ENABLE_AEAD_API_PREVIEW
   SRTO_CRYPTOMODE = 62,     // Encryption cipher mode (AES-CTR, AES-GCM, ...).
#endif
<snip>
```
If they are provided in a separate file, ex: srt-config.h app can also adapt their code to the lib they are building with.
```
#include <srt-config.h>
#include <srt.h>
...
```

An include for Makefiles can also be useful, to link with the libsrt configured crypto lib (--use-enclib=) for example.

**Comment by oviano** [comment]
I support this! I build SRT in a slightly unorthodox way by dropping the files directly into my already-existing cross-platform projects.

Having a <srt-config.h> file would be helpful in centralising the definitions common to all the builds, rather than have to add them individually to my projects.

**Comment by ethouris** [comment]
We could also make a config header file, but this solution only partially fixes any potential problems on this ground. A more general solution is to allow adding any kind of compile flags required for this header file before including, as obtained through `pkg-config`.

**Comment by jlsantiago0** [comment]
Yes. but then you would require pkg-config to access the data. It should always be available no matter how you access the library interface. and since the only way to ensure that you are interfacing with the information related to how the library was built, is to include it in generated .h files distributed with the library. Having to have magic #defines defined by the library user in their source or build environment when there is no sure way to know which ones are valid is not good.

**Comment by ethouris** [comment]
A small update - please tell me what you think.

We decided that the best option would be to completely free `srt.h` header (actually all SRT public headers) from any compile variant controlling macros. So, for example, all socket options will be available, at worst they just won't work, if the current variant configuration didn't enable particular feature at compile time.

The version that is prepared for this is an early stage candidate for the future 1.6.0 version, which is currently on the branch named `development`. For verification, this branch is also made a PR from, #3141. If you can, please check if this solves all problems as reported here.
